### PR TITLE
feat: added custom classnames for event-advanced tab

### PR DESCRIPTION
**Pull request by ibex088**
## What does this PR do?

This PR introduces customizable class names (customClassnames) for the Event Type advanced page in the project. This addition allows developers to style specific elements of the Event Type advanced page

#### Below is a full list of available custom class names with default styling:

```json
{
  "eventAdvancedTab": {
    "destinationCalendar": {
      "container": "bg-purple-50 border border-purple-200 shadow-md",
      "select": "border border-purple-300 shadow-sm",
      "innerClassNames": {
        "input": "",
        "option": "",
        "control": "",
        "singleValue": "",
        "valueContainer": "",
        "multiValue": "",
        "menu": "",
        "menuList": ""
      },
      "label": ""
    },
    "eventName": {
      "container": "bg-purple-50 border border-purple-200 shadow-sm",
      "input": "border border-purple-300 shadow-inner",
      "label": "",
      "addOn": ""
    },
    "addToCalendarEmailOrganizer": {
      "container": "bg-purple-50 border border-purple-200 shadow-inner",
      "label": "",
      "description": "",
      "children": "",
      "emailSelect": {
        "container": "",
        "select": "",
        "displayEmailLabel": ""
      }
    },
    "requiresConfirmation": {
      "container": "bg-purple-50 border border-purple-200 shadow-md",
      "label": "",
      "description": "",
      "children": "",
      "radioGroupContainer": "",
      "alwaysConfirmationRadio": "",
      "conditionalConfirmationRadio": {
        "container": "",
        "timeInput": "",
        "timeUnitSelect": "",
        "checkbox": "",
        "checkboxDescription": ""
      }
    },
    "bookerEmailVerification": {
      "container": "bg-purple-50 border border-purple-200 shadow-md",
      "label": "",
      "description": "",
      "children": ""
    },
    "calendarNotes": {
      "container": "bg-purple-50 border border-purple-200 shadow-md",
      "label": "",
      "description": "",
      "children": ""
    },
    "eventDetailsVisibility": {
      "container": "bg-purple-50 border border-purple-200 shadow-md",
      "label": "",
      "description": "",
      "children": ""
    },
    "bookingRedirect": {
      "container": "bg-purple-50 border border-purple-200 shadow-md",
      "label": "",
      "description": "",
      "children": "",
      "redirectUrlInput": {
        "container": "bg-purple-50 border border-purple-200 shadow-sm",
        "label": "",
        "input": "",
        "addOn": ""
      },
      "forwardParamsCheckbox": {
        "checkbox": "",
        "description": "",
        "container": "bg-purple-50 border border-purple-300 shadow-sm"
      },
      "error": "bg-purple-50 text-purple-700 border border-purple-300"
    },
    "seatsOptions": {
      "container": "bg-purple-50 border border-purple-200 shadow-md",
      "label": "",
      "description": "",
      "children": "",
      "seatsInput": {
        "container": "bg-purple-50 border border-purple-200 shadow-sm",
        "label": "",
        "input": "",
        "addOn": ""
      },
      "showAttendeesCheckbox": {
        "checkbox": "",
        "description": "",
        "container": ""
      },
      "showAvailableSeatCountCheckbox": {
        "checkbox": "",
        "description": "",
        "container": ""
      }
    },
    "timezoneLock": {
      "container": "bg-purple-50 border border-purple-200 shadow-md",
      "label": "",
      "description": "",
      "children": ""
    },
    "eventTypeColors": {
      "container": "bg-purple-50 border border-purple-200 shadow-md",
      "label": "",
      "description": "",
      "children": "",
      "warningText": ""
    },
    "roundRobinReschedule": {
      "container": "bg-purple-50 border border-purple-200 shadow-md",
      "label": "",
      "description": "",
      "children": ""
    },
    "emailNotifications": {
      "container": "bg-purple-50 border border-purple-200 shadow-md",
      "label": "",
      "description": "",
      "children": "",
      "confirmationDialog": {
        "container": "bg-purple-50 border border-purple-300 shadow-lg",
        "dialogTitle": "",
        "description": "",
        "confirmInput": {
          "container": "",
          "label": "",
          "input": "",
          "addOn": ""
        },
        "dialogFooter": {
          "container": "bg-purple-50",
          "confirmButton": "",
          "cancelButton": ""
        }
      }
    }
  }
}
```

#### Screenshot of Changes

Screenshot 2024-11-12 at 5 01 17 PM
Screenshot 2024-11-12 at 5 01 23 PM

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

**Review by zomars — DISMISSED** [review]
A couple comments.

-  Is this solving a specific issue? Can we link to it?
- We have several checks failing. Can we have a look at those?

File: packages/platform/examples/base/src/pages/event-types.tsx [hunk 1]
```diff
@@ -95,7 +95,161 @@ export default function Bookings(props: { calUsername: string; calEmail: string
...
+                    label: "",
+                  },
+                  eventName: {
+                    container: "bg-purple-50 border border-purple-200 shadow-sm",
+                    input: "border border-purple-300 shadow-inner",
+                    label: "",
+                    addOn: "",
+                  },
+                  addToCalendarEmailOrganizer: {
+                    switch: {
```
Comment (supalarry) [inline packages/platform/examples/base/src/pages/event-types.tsx:123]: Is there a reason `switch` is an object with container in it and other places we have `switchContainer` ?

Comment (SomayChauhan) [reply packages/platform/examples/base/src/pages/event-types.tsx]: Screenshot 2024-11-21 at 5 34 58 PM
fixed

**Comment by SomayChauhan** [comment]
Skipped form-builder(booking questions)for now, looked complex
Will get back to it in a follow up

**Review by supalarry — DISMISSED** [review]

**Review by supalarry — APPROVED** [review]

**Pull request merged**